### PR TITLE
RWA-2108: Error scenarios and clean up of request context for Granula…

### PR DESCRIPTION
**Pull request by rasikasri**
…r Permissions

**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/CftQueryServiceTest.java [hunk 1]
```diff
@@ -579,6 +579,48 @@ void shouldReturnRequestContextAvailableTasksOnly() {
             assertEquals("Hearing work", taskResourceList.getTasks().get(0).getWorkTypeLabel());
         }
 
+        @Test
+        void shouldReturnRequestContextAvailableTasksOnlyAndIgnoreSearchParameterAvailableTasksOnly() {
```
Comment (raghera) [inline src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/CftQueryServiceTest.java:583]: Minor point but would remove reference to `Only` when referring to the request context where possible.

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskSearchControllerTest.java [hunk 2]
```diff
@@ -1582,7 +1579,64 @@ void should_return_a_400_for_invalid_request_context() throws Exception {
...
+                status().isBadRequest(),
+                content().contentType(APPLICATION_PROBLEM_JSON_VALUE),
+                jsonPath("$.type")
+                    .value(
+                        "https://github.com/hmcts/wa-task-management-api/problem/bad-request"
+                    ),
+                jsonPath("$.title").value("Bad Request"),
+                jsonPath("$.status").value(400),
+                jsonPath("$.detail").value("Invalid request field: request_context")
+            ));
```
Comment (raghera) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskSearchControllerTest.java:1639]: Maybe there's another test somewhere I've missed but what if they send a non empty string in the request_context.  E.g. request_context=tasks_unavailable.  We should return a 400 error in this case.

File: src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/CftQueryServiceTest.java (hunk 1 above)
Comment (rasikasri) [reply src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/CftQueryServiceTest.java:583]: Changed to shouldReturnRequestContextAvailableTasksAndIgnoreSearchParameterAvailableTasksOnly, and some other places as well.

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskSearchControllerTest.java (hunk 2 above)
Comment (rasikasri) [reply src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskSearchControllerTest.java:1639]: Yes, there is this test in the same class - should_return_a_400_for_invalid_request_context

**Review by raghera — APPROVED** [review]
